six/modules/c++/six.sicd/source/CropUtils.cpp

```diff
@@ -166,6 +121,40 @@ namespace six
 {
 namespace sicd
 {
+
+PixelToLatLon::PixelToLatLon(const six::sicd::ComplexData& data,
+                             const scene::SceneGeometry& geom,
+                             const scene::ProjectionModel& projection) :
+    mGeom(geom),
+    mProjection(projection),
+    mOffset(data.imageData->scpPixel.row -
+                    static_cast<double>(data.imageData->firstRow),
+            data.imageData->scpPixel.col -
+                    static_cast<double>(data.imageData->firstCol)),
+    mSampleSpacing(data.grid->row->sampleSpacing,
+                   data.grid->col->sampleSpacing),
+    mGroundPlaneNormal(mGeom.getReferencePosition())
+{
+    mGroundPlaneNormal.normalize();
+}
+
+scene::LatLon PixelToLatLon::operator()(size_t row, size_t col) const
+{
+    const types::RowCol<double> imagePt(
```

Comment:
Take a look at `ComplexData::pixelToImagePoint()`.  Skimming over things, I'm pretty sure you could use that to get imagePt and remove `mOffset` and `mSampleSpacing` member variables (technically it's a little less efficient as you'd be re-computing some stuff but it'll be trivial compared to the `imageToScene()` call).

I added `pixelToImagePoint()` relatively recently... looks like I didn't notice that I could have updated this code to use it.

I do think `PixelToLatLon` should stay as its own class rather than be a member of `ComplexData` though so that `ComplexData` doesn't know about `ProjectionModel`.  I could be convinced otherwise but I think this standalone class is easier to use for this anyway.